common/src/test/java/apoc/ApocConfigTest.java

```diff
@@ -33,55 +41,158 @@
+        );
+    }
+
+    private void setApocConfigSystemProperty() {
+        System.setProperty(SUN_JAVA_COMMAND, "com.neo4j.server.enterprise.CommercialEntryPoint --home-dir=/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02 --config-dir=" +  apocConfigFile.getParent());
     }
 
     @Test
     public void testDetermineNeo4jConfFolderDefault() {
         System.setProperty(SUN_JAVA_COMMAND, "");
-        assertEquals(".", cut.determineNeo4jConfFolder());
+        assertEquals(".", apocConfig.determineNeo4jConfFolder());
     }
 
     @Test
     public void testDetermineNeo4jConfFolder() {
         System.setProperty(SUN_JAVA_COMMAND, "com.neo4j.server.enterprise.CommercialEntryPoint --home-dir=/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02 --config-dir=/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02/conf");
 
-        assertEquals("/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02/conf", cut.determineNeo4jConfFolder());
+        assertEquals("/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02/conf", apocConfig.determineNeo4jConfFolder());
     }
 
     @Test
-    public void testApocConfFileBeingLoaded() throws Exception {
-        String confDir = new File(getClass().getClassLoader().getResource("apoc.conf").toURI()).getParent();
-        System.setProperty(SUN_JAVA_COMMAND, "com.neo4j.server.enterprise.CommercialEntryPoint --home-dir=/home/stefan/neo4j-enterprise-4.0.0-alpha09mr02 --config-dir=" + confDir);
-        cut.init();
+    public void testApocConfFileBeingLoaded() {
+        setApocConfigSystemProperty();
+        apocConfig.init();
 
-        assertEquals("bar", cut.getConfig().getString("foo"));
+        assertEquals("bar", apocConfig.getConfig().getString("foo"));
     }
 
     @Test
     public void testDetermineNeo4jConfFolderWithWhitespaces() {
         System.setProperty(SUN_JAVA_COMMAND, "com.neo4j.server.enterprise.CommercialEntryPoint --config-dir=/home/stefan/neo4j enterprise-4.0.0-alpha09mr02/conf --home-dir=/home/stefan/neo4j enterprise-4.0.0-alpha09mr02");
 
-        assertEquals("/home/stefan/neo4j enterprise-4.0.0-alpha09mr02/conf", cut.determineNeo4jConfFolder());
+        assertEquals("/home/stefan/neo4j enterprise-4.0.0-alpha09mr02/conf", apocConfig.determineNeo4jConfFolder());
+    }
+
+    @Test
+    public void testApocConfWithExpandCommands() {
+        setApocConfigSystemProperty();
+        apocConfig.init();
+
+        assertEquals("expanded value", apocConfig.getConfig().getString("command.expansion"));
+    }
+
+    @Test
+    public void testApocConfWithInvalidExpandCommands() throws Exception {
+        setApocConfigSystemProperty();
+
+        String invalidExpandLine = "command.expansion.3=$(fakeCommand 3 + 3)";
+        addLineToApocConfig(invalidExpandLine);
+
+        RuntimeException e = assertThrows(RuntimeException.class, apocConfig::init);
+        String expectedMessage = "java.io.IOException: Cannot run program \"fakeCommand\": error=2, No such file or directory";
```

Comment:
Could we wrap this in a better exception so someone knows what config option failed to expand exactly?

Reply:
So this exception would be written in the logs on startup, which causes Neo to not start, and directly before the line:

`            log.info("Executing external script to retrieve value of setting " + settingName);
`

Gets printed, so the user should see that and the exception. Is that what you were thinking? I can also catch the exception and add more info, but then it will behave differently to Neo (this error comes from Neo)

Reply:
Yep, the current log is good enough then in my opinion